lib/node/nodes/line-sequential.js

```diff
@@ -43,3 +43,63 @@ var routingSequentialQueryTemplate = Node.template([
     '  {{? it.category_column }}GROUP BY {{=it.category_column}}{{?}}',
     ') _cdb_analysis_line_sequential'
 ].join('\n'));
+
+var countDistinctValues = Node.template([
+    'SELECT COUNT(*) AS count_distict_values FROM (',
+    '  SELECT DISTINCT {{=it.category_column}}',
+    '  FROM (',
+    '    {{=it.source}}',
+    '  ) AS _line_sequential',
+    ') AS _cdb_analysis_count_distinct'
+].join('\n'));
+
+LineSequential.prototype.computeRequirements = function(databaseService, limits, callback) {
+    var inputRows = this.source.estimatedRequirements.numberOfRows;
+    var numCategories = 1;
```

Comment:
Since: a) this code will be eventually replaced b) it seems to work c) the closure vars that bother you are encapsulated in a single function, I'll keep this as is, for fear of introducing new bugs. are you OK with it?